lib/routes/misskey/utils.ts

```diff
@@ -28,4 +30,42 @@ const parseNotes = (data, site) =>
         };
     });
 
-export default { parseNotes, allowSiteList };
+async function getUserTimelineByUsername(username, site) {
+    const search_url = `https://${site}/api/users/search-by-username-and-host`;
+    const cacheUid = `misskey_username/${site}/${username}`;
+
+    const account_id = await cache.tryGet(cacheUid, async () => {
+        const search_response = await got({
+            method: 'post',
+            url: search_url,
+            json: {
+                username,
+                host: site,
+                detail: true,
+                limit: 1,
+            },
+        });
+        const userData = search_response.data.filter((item) => item.username === username);
+
+        if (userData.length === 0) {
+            throw new Error(`username ${username} not found`);
+        }
+        return userData[0].id;
```

Comment:
Use [`.find()`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/find) instead of `filter()` with `[0]`